stdlib/public/core/StaticString.swift

```diff
@@ -182,7 +182,7 @@ public struct StaticString {
 #endif
         i += 1
       }
-      UTF8.encode(unicodeScalar, into: sink)
+      Unicode.UTF8.encode(unicodeScalar, into: sink)
```

Comment:
This `@_transparent` StaticString method is using deprecated APIs.

Reply:
The internal [`Unicode.Scalar.withUTF8CodeUnits(_:)`][1] method has been ABI-public since Swift 5.0, so I tried changing the else branch to:

```swift
return unicodeScalar.withUTF8CodeUnits { body($0) }
```

There was a failure in [test/IRGen/objc_super.swift:139:11][2]:

```
error: CHECK: expected string not found in input
// CHECK: @"$ss12StaticStringV14withUTF8BufferyxxSRys5UInt8VGXElFyAEcfU_"
          ^
<stdin>:2290:1: note: scanning from here
entry:
^
<stdin>:2308:41: note: possible intended match here
define linkonce_odr hidden swiftcc void @"$ss12StaticStringV14withUTF8BufferyxxSRys5UInt8VGXElFxAFXEfU_"(%swift.opaque* noalias nocapture sret %0, i64 %1, i64 %2, i8* %3, %swift.opaque* %4, %swift.type* %R) #0 {
```

I think the expected mangled name contains the [`let sink: (UInt8) -> Void`][3] closure of the existing implementation.

[1]: <https://github.com/apple/swift/blob/swift-5.0-branch/stdlib/public/core/UnicodeScalar.swift#L470-L482>
[2]: <https://github.com/apple/swift/blob/3e090b483352b3c6f889eec521605fda3ec1ab51/test/IRGen/objc_super.swift#L139>
[3]: <https://github.com/apple/swift/blob/3e090b483352b3c6f889eec521605fda3ec1ab51/stdlib/public/core/StaticString.swift#L177>